cmd/k8s-operator/operator.go

```diff
@@ -259,6 +259,7 @@ func runReconcilers(opts reconcilerOpts) {
 		Cache: cache.Options{
 			ByObject: map[client.Object]cache.ByObject{
 				&corev1.Secret{}:                            nsFilter,
+				&corev1.Service{}:                           nsFilter,
```

Comment:
This would break all existing Service and Ingress reconcilers as they will no longer be watching and reconciling Services at cluster scope (so no longer reconcile Tailscale Kubernetes Services or update Ingresses when backend Services change)
We should add a comment at the top of this cache explaining how it works - there is no need to add a filter here when starting to watch a new resource type _unless_ we need to watch it at namespace scope. Adding a filter here for a resource type will mean that any reconciler that watches those resource types will now only watch the resources at operator's namespace, so we have to be careful to not break existing things

Reply:
Sorry this was silly, I was just focused in on thinking about the NodePort services. I suppose there's no scenario where the unit tests would have broken right? Since this cache isn't getting initialised in them. Would be nice to find a way to ensure that we don't slip up on things like this and lead to things being blocked in CI.

Reply:
Not at all, it highlights that we need actual e2e tests